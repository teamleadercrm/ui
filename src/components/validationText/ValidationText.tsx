import React, { ReactNode } from 'react';

import { GenericComponent } from '../../@types/types';
import { BoxProps } from '../box/Box';
import ErrorText from './ErrorText';
import HelpText from './HelpText';
import SuccessText from './SuccessText';
import WarningText from './WarningText';

export interface ValidationTextProps extends Omit<BoxProps, 'children'> {
  error?: ReactNode;
  help?: ReactNode;
  inverse?: boolean;
  success?: ReactNode;
  warning?: ReactNode;
}

const ValidationText: GenericComponent<ValidationTextProps> = ({
  error,
  inverse,
  help,
  success,
  warning,
  ...others
}) => {
  if (error && typeof error !== 'boolean') {
    return (
      <ErrorText inverse={inverse} {...others}>
        {error}
      </ErrorText>
    );
  }

  if (warning && typeof warning !== 'boolean') {
    return (
      <WarningText inverse={inverse} {...others}>
        {warning}
      </WarningText>
    );
  }

  if (success && typeof success !== 'boolean') {
    return (
      <SuccessText inverse={inverse} {...others}>
        {success}
      </SuccessText>
    );
  }

  if (help) {
    return (
      <HelpText inverse={inverse} {...others}>
        {help}
      </HelpText>
    );
  }

  return null;
};

export default ValidationText;
