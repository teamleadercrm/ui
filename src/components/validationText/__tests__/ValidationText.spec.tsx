import React from 'react';
import { render } from '@testing-library/react';
import ValidationText from '../ValidationText';

describe('Component - ValidationText', () => {
  it('renders nothing when no text is passed', async () => {
    const { asFragment } = render(<ValidationText />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a success text', async () => {
    const { asFragment } = render(<ValidationText success="Success text" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders an error text', async () => {
    const { asFragment } = render(<ValidationText error="Error text" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a warning text', async () => {
    const { asFragment } = render(<ValidationText warning="Warning text" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a help text', async () => {
    const { asFragment } = render(<ValidationText warning="Help text" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
