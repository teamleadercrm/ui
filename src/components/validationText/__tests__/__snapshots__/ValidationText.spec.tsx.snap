// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - ValidationText renders a help text 1`] = `
<DocumentFragment>
  <p
    class="box margin-top-1 text text-small dark gold"
    data-teamleader-ui="warning-text"
  >
    Help text
  </p>
</DocumentFragment>
`;

exports[`Component - ValidationText renders a success text 1`] = `
<DocumentFragment>
  <p
    class="box margin-top-1 text text-small dark mint"
    data-teamleader-ui="success-text"
  >
    Success text
  </p>
</DocumentFragment>
`;

exports[`Component - ValidationText renders a warning text 1`] = `
<DocumentFragment>
  <p
    class="box margin-top-1 text text-small dark gold"
    data-teamleader-ui="warning-text"
  >
    Warning text
  </p>
</DocumentFragment>
`;

exports[`Component - ValidationText renders an error text 1`] = `
<DocumentFragment>
  <p
    class="box margin-top-1 text text-small dark ruby"
    data-teamleader-ui="error-text"
  >
    Error text
  </p>
</DocumentFragment>
`;

exports[`Component - ValidationText renders nothing when no text is passed 1`] = `<DocumentFragment />`;
